### ISSUE TITLE
Why must pywebviewbe run on a main thread?

### ISSUE THREAD

**Issue by MisterSilvereagle** [r0x0r/pywebview#1251]
### Specification
- pywebview version: general
- operating system: general
- web renderer: general

### Description

  When starting webview in a different thread, it throws an exception saying `"pywebview must be run on a main thread."`
  This can be circumvented by either modifying the pywebview code or just renaming the thread that webview is running in to `"MainThread"`, as `__init__.py` only checks for `threading.current_thread().name`. 
  My question now is _not_ to fix this, but rather _why it is not allowed_. I simply don't get it, why I'm officially not allowed to run the gui in another thread. I tested it and noticed no issues by not running the gui in the main thread.

  **Why, if it is possible, is it still not allowed to run webview in a thread different to MainThread???**

### Practicalities

- YES I am willing to work on this issue myself(, but there is probably nothing to do.)

- NO I am prepared to support this issue financially.

**Comment by r0x0r** [comment]
Cocoa requires ui to be run on the main thread. As far as I know there is no way around it.
I do not want to remove the main thread requirement for other platforms, as it would break cross-platform compatibility.

As this feature request pops up regularly, I may introduce `webview.start(blocking=False)` that would start _pywebview_ on a secondary thread (another process on Cocoa) and as the result would not block the main thread.

**Comment by MisterSilvereagle** [comment]
Okay, I get that it may be a requirement on other platforms that I did not test. To make my project cross-platform-compatible, I will probably just start webview in another process myself.

I looked into the code, and it is not as simple as I thought it would be to implement this. If I have the time, I will make a pull request. But I can't guarantee anything.

Anyways, thanks for your reply, appreciate your great work on this project!
